### ISSUE TITLE
Reconsidering read/unread distinction

### ISSUE THREAD

**Issue by marienfressinaud** [FreshRSS/FreshRSS#2018]
Over the years, I realized that people were using the distinction between read and unread articles to keep a list of articles they wanted to read later. I honestly never thought of it this way and I personally use the "favorite" articles as a way to list articles that I want to read later (but I realize that it's also a diversion of functionality).

The problem is that it's very easy to lose all its unread articles (there are two buttons, and the warning is regularly skipped :p). The "favorite" feature is less problematic but might not fit with all the use cases.

It would help a lot if people who don't mark all their articles as read regularly can explain what they try to achieve!

**Comment by lapineige** [comment]
(special mention to this number 2018 ;)

**Comment by lapineige** [comment]
> It would help a lot if people who don't mark all their articles as read regularly can explain what they try to achieve!

For my part, I receive too much articles (it's kind of intended) to read them each day - as I'm not always able to read them on a daily (sometimes weekly) basis. But I often have little time in a day to sort them (without reading too much). Then I have a selection that is ready to be read later.

That's a short/mid-term solution. On the long term, I keep the unread articles (that might be interesting - as explained I sort them before) in case I'm searching something one day, to know if I've read the article (I might remember reading this information somewhere in my RSS feed) or not (that's something new).

**Comment by Alkarex** [comment]
By default, the option *mark as read during scroll* is enabled, in particular to reduce this behaviour of not marking as read.

**Comment by lapineige** [comment]
I think the current behaviour is pretty good. I'm just using it in a different way, it's more a diversion of functionality, and maybe an niche use case.
(I'm also using Wallabag as a read-it-later service, but it's an extra software)

**Comment by GLLM** [comment]
Hi !
I'm also using the unread status to keep all articles I want to read but do it later ...
I mark as read all articles I already have read and all those I find not interesting.
And I keep the longer ones unread to see them later.
Once read, if an article was really worth it and I want to keep it, I'll mark it as Favorite !

I'm always very super mega careful when I mark a selection of items as read since I'm always afraid to mark my selection of to-read-later items as unread unwillingly !!

 
**Edit**  : actually, what could be great could be that the articles can be marked as to-read-later ... which would keep them unread until individually marked as read.
This means mark all as read would not affect them ... 

### Workflow
Article arrives in feed:  it is unread.
Then: 
1. I find it useless, I mark it as read
2. I like it and read it : I mark it as read
3. I like it but i dont have enough time to read it now & I want to ensure I cannot loose it (i.e. mark it as read by mistake) : I mark it as -to-read-later. : this way it is not marked as read if I use the "mark all as read" buttons. When I later read it, I'll mark it as read

_In addition :_ 
What would be most useful (and there is an issue for this), is the ability to mark items as read without opening them ... since I'm able to know if I will read them or not based on the title.

**Comment by Frenzie** [comment]
@marienfressinaud 
> It would help a lot if people who don't mark all their articles as read regularly can explain what they try to achieve!

That's pretty simple. The whole purpose for me is not to miss things. For example, I subscribe to several dozen channels on YouTube, and I simultaneously subscribe to a couple of those in FreshRSS because on YouTube blink and you miss them. For most stuff I subscribe to on YT, that's acceptable to me since it's just something I might watch on the side while washing dishes.

I do use MAAR on select feeds occasionally, but if I were actually doing that regularly the feed in question would go on the shortlist for permanent removal.

The problem with the desktop feed readers I used previously is twofold:

1. They're bound to my desktop, but unlike in '02/'03ish when I enthusiastically started using feeds, I've now also been reading things on my cellphone for years. But even since a few years before then, since '07, I've had both a laptop *and* a desktop. Naturally I want all of my progress and configuration to carry over across devices.
2. They only collect while my desktop is running.

Out of the dozen or so (self-)hosted alternatives I tried, I thought FreshRSS was easily the best. I didn't just evaluate what I thought of the program out of the box, but also how easily I could hack it to behave the way I wanted something it didn't do. So out of several superficially similar projects I opted for FreshRSS based mainly on my impression of its code and community. Yet there's a minor but real disconnect between this weird Google Reader-style "oh look you didn't read this feed for 30 days so now you're out of luck 'cause it's gone forever" — that's how it felt when I off-handedly tried Google Reader back in the day ('05?) — and my basic more e-mail client-like desires. I set the purge setting to `1200` months (~100 years). ;-)

I can certainly imagine "useless" feeds that can be purged every couple of days (i.e., almost every website that posts more than one thing a day), but I don't subscribe to those in the first place. I just visit those sites when I feel like it. Interestingly, FreshRSS only has a "minimum number of articles to keep" setting, so I'd have to set that to something really high on everything combined with a low default to experiment with this.

So what am I actually missing, if anything? Deletion, perhaps. In readers like Opera M2 and QuiteRSS, there's favorite, read, unread, and gone. In FreshRSS, I just pressed <kbd>k</kbd> five times, because yes, I might still want to read more articles out of that book from last year. But the "read" view is pretty much useless in FreshRSS without adding favorites or using the search function. Whereas in something like QuiteRSS "read" is actually a useful archive all by itself, even without starring anything.

![screenshot_2018-09-11_21-10-05](https://user-images.githubusercontent.com/202757/45381975-2c5b3000-b607-11e8-943d-b488da141fa1.png)

In QuiteRSS, I'd select the same bunch and press <kbd>Del</kbd>.

![screenshot_2018-09-11_21-08-30](https://user-images.githubusercontent.com/202757/45381885-e43c0d80-b606-11e8-8bb3-e0fa7879f86e.png)

@GLLM 
> actually, what could be great could be that the articles can be marked as to-read-later ... which would keep them unread until individually marked as read.

I would just want to auto-mark everything as to-read-later until I explicitly dismiss it. :-)

> What would be most useful (and there is an issue for this), is the ability to mark items as read without opening them ... since I'm able to know if I will read them or not based on the title.

You can using the icon, and I think we recently added a new shortcut for doing something like that with the keyboard as well.

**Comment by Alkarex** [comment]
I use a little trick to get an additional state:
1. I mark as favourites the articles that seem to be interesting (which I often want to read in details later, and do not want to miss)
2. I register by RSS to my own favourites. This makes a feed which contain all the articles I have put in favourites.
3. I remove the articles from the favourites when read / processed
4. If very interesting, post to my Shaarli

**Comment by GLLM** [comment]
@Frenzie : I was indeed talking about a keyboard shortcut to do so. And when I said "there is an issue for this" I meant that someone already logged a Feature request in github issues.

@Alkarex : nice trick, I had not thought about this one :-)

**Comment by typhoe** [comment]
Hi,

just did a migration from TT-RSS to FreshRSS, and so far really liking this new RSS aggregator.

I'm browsing the issues to find an answer to this use case: When reading new posts and all my feeds are read, I don't want to default to display the already read posts. 
I would like to stay with nothing displayed (no unread posts), which seems to be the url options "i/?a=normal&state=2".

Is there a way to obtain this behavior?

Otherwise, I use read/unread/favorite kind of like @Alkarex (except the trick about registering my own favorites as a rss feed, which is a nice trick by the way!).

Another question if I may: TT-RSS allows to export its Favorites (xml file)... any way to import it in FreshRSS?

Thank you

**Comment by Alkarex** [comment]
@typhoe Hello and welcome :-),
The option you are asking for is under Reading / Articles to display:
![image](https://user-images.githubusercontent.com/1008324/47444120-1029e180-d7b6-11e8-99c4-3f4a45350601.png)

FreshRSS can import standard OPML, but for favourites, it depends on the format. Could you please send me an example from TT-RSS? Either here or [by e-mail](http://alexandre.alapetite.fr/cv/).

**Comment by typhoe** [comment]
Indeed, thank you!
I set my to "adjust showing" and didn't notice the other option.

I'll send an example by email for you to check if the import could be possible.
Thank you again!

**Comment by Frenzie** [comment]
Quoting @Alkarex from https://github.com/FreshRSS/FreshRSS/issues/1167#issuecomment-480574486

> With @Frenzie 's example:
> * See an article to read later -> star or label
> * See an article to keep -> star or label
> * Unsure -> mark as unread
> * Read / delete -> mark as read

That's the problem in a nutshell. You're suggesting all kinds of actions where there should be none. :-)

* Read later → do nothing (or mark as unread if you accidentally marked it as read)
* Favorite → favorite (rarely used)
* Keep → do nothing
* Delete → delete

Read/unread has nothing to do with deletion. The only deletions that should ever occur are those explicitly mandated by me. No data loss. To be clear, all of this already works fine. The only thing missing is deletions.

Rebranding existing actions is no solution, because you'd still be missing a state:

* read → (soft) delete
* unread → read
* favorite → unread
* super deluxe favorite → favorite.

That being said, I could easily live without integrated favorites. The main problem there is that you'd completely change the way the program works, and it'd definitely make API compatibility harder. By constrast, a soft delete is API-transparent because it simply wouldn't know about it. (Incidentally, the tt-rss API might support it directly.)

PS I vehemently disagree with @lapineige that this is in any way "niche." This is how every newsreader and e-mail client works, except those inspired by "broken by design" Google Reader.

**Comment by Alkarex** [comment]
I am truly making an effort to understand the details, so please bear with me if I am a bit slow :-)
You do seem to ask for an additional state, right?

**Comment by Frenzie** [comment]
Yes, the deleted state as described here by @aledeg: https://github.com/FreshRSS/FreshRSS/issues/1167#issuecomment-480534184.

**Comment by Alkarex** [comment]
Right, then this is not compatible with our APIs and clients, since there is no such concept or action. Am I missing something? How would our clients tell what article is read and to be kept vs. read and for deletion?
I think my confusion comes from the reading of contradictory things :-P

**Comment by Frenzie** [comment]
It would be completely transparent to clients, but it's indeed an incomprehensible[1] design flaw in Google Reader and its API.

There wouldn't be any discernible change with the current situation for the API. How does the API take care of a setting like `Remove articles after X months`? It doesn't.

If that setting causes an article to be marked as deleted instead of deleting it straight away, you'd have a few more weeks to notice and undo it. That would be slightly more user-friendly but otherwise it's a difference without distinction. Whatever currently happens API-wise still happens.

[1] I suspect it's a typical Google-centric design, meaning the Google Reader server remembers everything anyway and damn the user.

**Comment by Frenzie** [comment]
PS I'd want never to be an option for the auto-deletion setting, instead of the workaround of setting it to some impossibly high value, but that aside. Point being that, within the context of the Google Reader API, deletion is a FreshRSS affair no matter the implementation specifics.

**Comment by Alkarex** [comment]
Related discussion https://github.com/FreshRSS/FreshRSS/issues/493 (old and in French)

**Comment by Alkarex** [comment]
@Frenzie I am afraid I have still not understood, and I hope you have a bit of patience left :-P
If there is no possibility for marking an article for deletion, how does this additional state makes any difference? Or are you saying that you want this feature only in the Web version? And in the Web version, what would the UI look like?
We already have a setting to keep old articles for a potentially long time before removing them, and the possibility to keep articles forever if they have a star or label. So we do not delete articles straight away.

**Comment by Frenzie** [comment]
> We already have a setting to keep old articles for potentially a long time before removing them, and the possibility to keep articles forever if they have a star or label. So we do not delete articles straight away.

Right, but this is backwards. Nothing should be automatically deleted, yet a variety of items should be manually removable to prevent archive pollution.

> Or are you saying that you want this feature only in the Web version?

Well of course not, but that's necessarily how it must be while operating within the constraints of the Google Reader API.

> And in the Web version, what would the UI look like?

<kbd>Del</kbd>

And yes, there would also have to be an optional red X, a paper shredder or a trash can icon of some sort.

<hr>

For example, if I search for `git workflow`, you can see the pollution marked with a red cross. This is the content that after reading or skimming, I know I will never want to see again. (Based in this case mainly on the title, so possibly slightly wrong.) And yet, there's only *one* must-read reference article among the bunch. The rest is there just in case. It's part search optimization and part memory security. In a few years, half this stuff will probably have disappeared from the web.

![Screenshot_2019-04-07_16-51-28](https://user-images.githubusercontent.com/202757/55685375-aeb99180-5955-11e9-8f3f-2eaa91d07516.png)

An approximation of the desired behavior can be easily effected through the GUI since #2027 by labeling articles something like `delete` and letting a cron job torpedo the older ones with a simple SQL query. (Whether the articles are shown in unread view doesn't concern me much, but some JS could take care of that right now as well.) Come to think of it, such an approach might even work with the Google Reader API, although most clients don't seem to implement labels.

**Comment by math-GH** [comment]
That is a very long discussion.

Could someone please summarize it?

**Comment by Frenzie** [comment]
1. To some people, a newsreader is something that rotates, a bit like if you picked up today's newspaper. Old stuff is irrelevant and can be ignored/discarded. Favoriting is the mechanism to keep the occasional item.

2. To others, a newsreader is roughly the same as an email client. The only real difference is that email is something that requires action while feeds can safely be ignored, hence why they should be separated. But other than that they should behave more or less identically. Deletion is very purposeful and you should never have to mark something to be kept.

<hr>

1. Is exemplified by a feed like the BBC. I would never subscribe to that, but simply visit the website if I wanted to see what's up. Or possibly run a completely separate instance set up a different way.

2. Is basically all feeds that publish something once or twice a day at most (or possibly several things once or twice a week).

<hr>

The reason we 2 people use a server-based app like FreshRSS is because a desktop program like QuiteRSS/Vivaldi doesn't always run and therefore you might miss things, plus it's not accessible/synced on your cellphone or other computers. But fundamentally, we consider 1 to be broken by design.

**Comment by marienfressinaud** [comment]
I think to remember that my initial idea was similar to the read-later mechanism of @GLLM (I probably had in mind to rename "favorite" to "read later"? but it's not a good solution). While I understand your logic @Frenzie (and I agree that being able to remove some articles would be useful), I'm not sure it answers the initial issue: we may lose the unread status of a lot of articles by hitting the "mark all as read" button by accident.

In the mean time, the current situation seems to please most of the people, so maybe there's nothing to do? :) Except for those who are afraid to hit the "mark all as read" button by accident, maybe hide it with CSS? :D

**Comment by Alkarex** [comment]
> Except for those who are afraid to hit the "mark all as read" button by accident

We also have a confirmation options :-)

**Comment by marienfressinaud** [comment]
I know for a fact that, sometimes, the confirmation option is not enough :p

**Comment by Frenzie** [comment]
@marienfressinaud 

> While I understand your logic @Frenzie (and I agree that being able to remove some articles would be useful), I'm not sure it answers the initial issue: we may lose the unread status of a lot of articles by hitting the "mark all as read" button by accident.

Er, your parenthetical remark is kind of the exact opposite of what I said? :-)

**Comment by marienfressinaud** [comment]
:thinking: I was referring to https://github.com/FreshRSS/FreshRSS/issues/2018#issuecomment-420393758

> So what am I actually missing, if anything? Deletion, perhaps. […]

And all the discussion that followed

**Comment by Nono-m0le** [comment]
Sorry to hijack the thread but this would definitelly solve my current issue to migrate to FreshRRS.
Would be really nice to have a shortcut "open the next article, WITHOUT marking the current as read" (aka, "keep the current one as unread, I'll spend more time later on"). I don't like to fill up the "favorites" label for this, but just keep the unread articles on the list of (new) feed.

**Comment by Alkarex** [comment]
@Nono-m0le In your reading settings, have you tried un-checking the following option?

![image](https://user-images.githubusercontent.com/1008324/205070304-be1f6382-b056-4cb5-8426-0be2d0974fb5.png)

Then you can use shortcuts to explicitly mark as read, and jump to the next article without marking it as read.

A slightly related little hint: if you use *mark as read while scrolling* and you manually mark an article as unread (or with shortcut), it will remain unread during that scroll session

**Comment by Nono-m0le** [comment]
I did ticked it, but I'm not sure I should or not.
The ideal scenario would be to **not** mark as read anything, untill I explicitely use a shortcut.
I'm use to have/use only 3 shortcuts : 
One for "marking the current as read, and put the focus on the next article"
Two for "Switching to the next article, WITHOUT marking the current as read"
Three for "Open the page externally"

It this already possible with the options ?

**Comment by Frenzie** [comment]
Yes, in reading → mark article as read. By default this setting is enabled.

![image](https://user-images.githubusercontent.com/202757/205259986-8be8f936-ca70-4819-87b6-5958af4f09a8.png)

**Comment by Nono-m0le** [comment]
I'm sorry in advance if this thread looks like a support ticket for dumb user like me but, If I understand correctly, with this settings : 
![image](https://user-images.githubusercontent.com/3726435/205260364-7c67ec18-e6ca-4f70-8170-fd1fe1cd75af.png)
I do : 
"j" for my "marking the current as read, and put the focus on the next article" aka: 
![image](https://user-images.githubusercontent.com/3726435/205260704-242b7034-b12d-41c9-af9f-9954e258e6af.png)
I do: 
"n" for my "Switching to the next article, WITHOUT marking the current as read" aka: 
![image](https://user-images.githubusercontent.com/3726435/205260800-f709301c-af8e-4801-8a28-938e4ca1ede4.png)
and obviously "space" for my "Open the page externally" aka: 
![image](https://user-images.githubusercontent.com/3726435/205260984-80e24a96-90ca-4fc8-95a1-dfed58066976.png)

If that's correct, when I'm on this particular feed (as an example) : 
![image](https://user-images.githubusercontent.com/3726435/205261445-6ef1053c-b244-4e09-b8cf-6a1b94b8de49.png)

When I'm pressing "j" I would assume, mark the current article as read, and put the focus on the 2nd article. But on this case, the current article is still "visible" (but mark as read, which is correct) and the focus is NOT on the 2nd : 

![image](https://user-images.githubusercontent.com/3726435/205261591-8eb20137-fd9c-44b4-89d2-83009fdad6ac.png)

It seems to be even "worst" when I'm doing "n" as the article is ALSO back in white, which indicate that it has been marked as read ? as seen here : 
![image](https://user-images.githubusercontent.com/3726435/205262213-e0475dae-57b3-4d6c-aeb6-15b9a4301164.png)

**Comment by Nono-m0le** [comment]
In case it's needed, I'll be happy to do a screensharing session to share the behavior i'm facing ?

**Comment by Frenzie** [comment]
> It seems to be even "worst" when I'm doing "n" as the article is ALSO back in white, which indicate that it has been marked as read ? as seen here :

The closed envelope and the red border on the left indicate that it's unread, though normally I believe the header does indicate that as well. Maybe some custom CSS or a theme that's a bit different?

**Comment by Nono-m0le** [comment]
it's a default/new installation. No specific customization at my end.

**Comment by petesimonic** [comment]
I can understand how this issue was raised, I see that it's alive, but I can't understand what exactly is going on, and what outcome is currently expected out of these 5 years of its life. X)

Is it possible to sketch out current state? %) Would another user-feedback add something?

**Comment by Nono-m0le** [comment]
I could arrange a video meeting (maybe even recording) if that may help...

**Comment by petesimonic** [comment]
Looks like a step to "sketching out" to me! X)
> I could arrange a... (...recording) ...

Furthermore most of video talks are quite straightforward converted to text. So it's very probable that your talk would let us have written summary/status too!

**Comment by Nono-m0le** [comment]
Do you know any (free / opensource) tool that would do that by any chance ?
I hope you don't mind the French accent :D

**Comment by petesimonic** [comment]
How do you like 
[VLC](https://superuser.com/questions/612186/how-to-capture-screen-video-with-audio-using-vlc-media-player)?
> Do you know any (free / opensource) tool that would do that by any
> chance ?

Sounds good. X)
